### ISSUE TITLE
Is there  a way to use proxy for dredd?

### ISSUE THREAD

**Issue by alex-glv** [apiaryio/dredd#591]
I want to be able to configure proxy for dredd to be able to reach my endpoint.
Is there currently a way to do it or should I set an nginx service to proxy requests?

**Comment by honzajavorek** [comment]
I don't think there's any way, currently, but maybe @netmilk knows about something.

**Comment by alex-glv** [comment]
Thanks, @honzajavorek 

Apparently, the difference between proxied and non-proxied requist is the destination host and HTTP request.

Say, your proxy is at ```http://proxy``` and your API endpoint is at ```http://api```.

To proxy the request, http request should look s follows:
```
GET http://api/v1/endpoint
Host: http://proxy
[...]
```

The requests will go through the proxy server.

So, finally to use proxy from dredd one has to insert a hook into e.g hooks.js file:
```
hooks.beforeEach((transaction, done) => {
    transaction.fullPath = "http://api" + transaction.fullPath;
    [...]
});
```

And issue Dredd command with the  proxy as the destination:
```
dredd swagger-spec.json http://proxy  -f hooks.js
```

**Comment by honzajavorek** [comment]
Hmm, this could go to docs 📖

**Comment by honzajavorek** [comment]
@alex-glv Closing this as a duplicate of https://github.com/apiaryio/dredd/issues/51. I won't put your example to docs, because I'm going to work on a proxy support in Dredd 🙂 I'd appreciate if you watch out for new versions of Dredd and once the feature is released, if you could let me know it solves your problem.

**Comment by honzajavorek** [comment]
Support for HTTP(S) proxy has been implemented in Dredd [v3.3.0](https://github.com/apiaryio/dredd/releases/tag/v3.3.0). See [documentation](https://dredd.readthedocs.io/en/latest/how-it-works/#using-https-proxy) on how to use it.
